### PR TITLE
Adds opt-in quick_discouraged_focused_test rule

### PR DESCRIPTION
**Pull request by ornithocoder**
Implements #1905.

### CODE REVIEW

File: Source/SwiftLintFramework/Rules/QuickDiscouragedFocusedTestRule.swift [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        return [StyleViolation(ruleDescription: type(of: self).description,
+                               severity: configuration.severity,
+                               location: Location(file: file, byteOffset: offset))]
+    }
+}
+
+private enum QuickFocusedCallKind: String {
+    case fdescribe
+    case fcontext
+    case fit
```
Comment (marcelofabri) [inline Source/SwiftLintFramework/Rules/QuickDiscouragedFocusedTestRule.swift]: should this also catch `fitBehavesLike`?

Comment (ornithocoder) [reply Source/SwiftLintFramework/Rules/QuickDiscouragedFocusedTestRule.swift]: Definitely! ;-)

Comment (ornithocoder) [reply Source/SwiftLintFramework/Rules/QuickDiscouragedFocusedTestRule.swift]: I've also added it as a valid keyword in #1904. Thanks 👍

**Comment by jpsim** [comment]
LGTM! Thanks for the PR 👍

**Pull request merged**